src/main/java/com/fundynamic/d2tm/game/entities/EntityRepository.java

```diff
@@ -0,0 +1,138 @@
+package com.fundynamic.d2tm.game.entities;
+
+
+import com.fundynamic.d2tm.game.entities.structures.Structure;
+import com.fundynamic.d2tm.game.entities.units.Unit;
+import com.fundynamic.d2tm.game.map.CellAlreadyOccupiedException;
+import com.fundynamic.d2tm.game.map.Map;
+import com.fundynamic.d2tm.math.Vector2D;
+import org.newdawn.slick.Image;
+import org.newdawn.slick.SlickException;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+public class EntityRepository {
+
+    public static int TRIKE = 0;
+    public static int QUAD = 1;
+
+    public static int CONSTRUCTION_YARD = 0;
+    public static int REFINERY = 1;
+
+    private final Map map;
+
+    private HashMap<String, EntityData> entitiesData;
+
+    private Set<Entity> entities;
+
+    public EntityRepository(Map map) throws SlickException {
+        this(map, new HashMap<String, EntityData>());
+
+        // TODO: read this data from an external (XML/JSON/YML/INI) file
+        createUnit(QUAD, "units/quad.png", 32, 32);
+        createUnit(TRIKE, "units/trike.png", 28, 26);
+        createStructure(CONSTRUCTION_YARD, "structures/2x2_constyard.png", 64, 64);
+        createStructure(REFINERY, "structures/3x2_refinery.png", 96, 64);
+    }
+
+    public EntityRepository(Map map, HashMap<String, EntityData> entitiesData) throws SlickException {
+        this.map = map;
+        this.entitiesData = entitiesData;
+        this.entities = new HashSet<>();
+    }
+
+    public void placeUnitOnMap(Vector2D topLeft, int id) {
+        placeOnMap(topLeft, EntityType.UNIT, id);
```

Comment:
I think it's weird that the ID and type are treated separately. I'd rather see something like this:

```java
public class EntityBlueprint {
  public int id;
  public EntityType type;

  public EntityBlueprint(int id, EntityType type) {
    this.id = id;
    this.type = type;
  }
}

public class EntityRepository {
  public static final EntityBlueprint TRIKE = new EntityBlueprint(0, EntityType.Unit);
  public static final EntityBlueprint QUAD = new EntityBlueprint(1, EntityType.Unit);

  public static final EntityBlueprint CONSTRUCTION_YARD = new EntityBlueprint(0, EntityType.Structure);
  public static final EntityBlueprint REFINERY = new EntityBlueprint(1, EntityType.Structure);

  ... <snip> ...
}
```

Or do you think that's overkill? Will we even keep these statics eventually or is it all going to be configurable?